### PR TITLE
Multi Campaign Support

### PR DESCRIPTION
**Pull request by linkielink**
This PR adds multi campaign support for assets

### CODE REVIEW

File: src/components/Modals/AssetAmountSelectActionModal.tsx [hunk 1]
```diff
@@ -84,22 +84,30 @@ export default function AssetAmountSelectActionModal(props: Props) {
...
+                    textClassName='text-white'
+                    size='sm'
+                    amount={updatedAmount}
+                    withLogo
+                  />
+                </div>
+              )
+            return null
+          })
+        : null}
```
Comment (StefChatz) [inline src/components/Modals/AssetAmountSelectActionModal.tsx:110]:
```suggestion
      {showCampaignHeader && asset.campaigns
        && asset.campaigns.map((campaign, index) => {
            if (campaign.type === 'points_with_multiplier')
              return (
                <div
                  className={classNames(
                    'w-full p-2 flex items-center justify-center',
                    campaign?.bgClassNames ?? 'bg-white/50',
                  )}
                  key={index}
                >
                  <AssetCampaignCopy
                    campaign={campaign}
                    asset={asset}
                    textClassName='text-white'
                    size='sm'
                    amount={updatedAmount}
                    withLogo
                  />
                </div>
              )
            return null
          })
        }
```

File: src/components/Modals/AssetsSelect/index.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ export default function AssetsSelect({
         const balancesForAsset = balances.find(byDenom(asset.denom))
         const coin = BNCoin.fromDenomAndBigNumber(asset.denom, BN(balancesForAsset?.amount ?? '0'))
         const value = getCoinValue(coin, assets)
-        asset.campaign = isBorrow ? undefined : asset.campaign
+        asset.campaigns = isBorrow ? undefined : asset.campaigns
```
Comment (StefChatz) [inline src/components/Modals/AssetsSelect/index.tsx:46]:
```suggestion
        asset.campaigns = isBorrow || asset.campaigns
```

File: src/components/Modals/Farm/FarmModalContentHeader.tsx [hunk 3]
```diff
@@ -105,22 +107,30 @@ export default function FarmModalContentHeader(props: Props) {
...
+                    size='sm'
+                    campaign={campaign}
+                    amount={updatedDepositedValue}
+                    withLogo
+                  />
+                </div>
+              )
+            return null
+          })
+        : null}
```
Comment (StefChatz) [inline src/components/Modals/Farm/FarmModalContentHeader.tsx:133]:
```suggestion
      {showCampaignHeader && poolToken.campaigns
        && poolToken.campaigns.map((campaign, index) => {
            if (campaign.type === 'points_with_multiplier')
              return (
                <div
                  className={classNames(
                    'w-full p-2 flex items-center justify-center',
                    campaign?.bgClassNames ?? 'bg-white/50',
                  )}
                  key={index}
                >
                  <AssetCampaignCopy
                    asset={poolToken}
                    textClassName='text-white'
                    size='sm'
                    campaign={campaign}
                    amount={updatedDepositedValue}
                    withLogo
                  />
                </div>
              )
            return null
          })
        }
```

File: src/components/account/AccountFund/AccountFundRow.tsx [hunk 4]
```diff
@@ -32,17 +32,21 @@ export default function AccountFundRow(props: Props) {
...
+                asset={asset}
+                size='sm'
+                amount={props.amount}
+                withLogo
+                className='justify-center'
+                key={index}
+              />
+            </div>
+          ))
+        : null}
```
Comment (StefChatz) [inline src/components/account/AccountFund/AccountFundRow.tsx:49]:
```suggestion
      {asset.campaigns && asset.campaigns.length >= 1
        && asset.campaigns.map((campaign, index) => (
            <div className='flex justify-center w-full p-2 mt-4 border rounded border-white/20'>
              <AssetCampaignCopy
                campaign={campaign}
                asset={asset}
                size='sm'
                amount={props.amount}
                withLogo
                className='justify-center'
                key={index}
              />
            </div>
          ))
        }
```

File: src/components/earn/farm/common/Table/Columns/Name.tsx [hunk 5]
```diff
@@ -59,13 +59,17 @@ export default function Name(props: Props) {
...
+            ? poolAsset.campaigns.map((campaign, index) => (
+                <AssetCampaignCopy
+                  asset={poolAsset}
+                  campaign={campaign}
+                  size='xs'
+                  amount={lpAmount.isZero() ? undefined : lpAmount}
+                  key={index}
+                />
+              ))
+            : null}
```
Comment (StefChatz) [inline src/components/earn/farm/common/Table/Columns/Name.tsx:72]:
```suggestion
          {poolAsset?.campaigns && poolAsset.campaigns.length >= 1
            && poolAsset.campaigns.map((campaign, index) => (
                <AssetCampaignCopy
                  asset={poolAsset}
                  campaign={campaign}
                  size='xs'
                  amount={lpAmount.isZero() ? undefined : lpAmount}
                  key={index}
                />
              ))
            }
```

File: src/components/earn/lend/Table/Columns/Name.tsx [hunk 6]
```diff
@@ -23,11 +21,22 @@ export default function Name(props: Props) {
...
+                      asset={props.asset}
+                      size='xs'
+                      noDot
+                      amount={props.amount}
+                      key={index}
+                    />
+                  </>
+                ))
+              : null
+            : asset.name
```
Comment (StefChatz) [inline src/components/earn/lend/Table/Columns/Name.tsx:39]: Is it best practice to have double ternary operations?

**Review by StefChatz — CHANGES_REQUESTED** [review]

**Comment by linkielink** [comment]
Note to myself: never create a hasted PR again.

**Review by StefChatz — APPROVED** [review]

**Pull request merged**